docs/llama_multi_lora_tutorial.md

```diff
@@ -0,0 +1,242 @@
+# Depolying multi-lora vLLM backend in Triton
+
+The idea of multi-lora was proposed recently, for more please refer to:
+
++ [S-LoRA: Serving Thousands of Concurrent LoRA Adapters](https://arxiv.org/abs/2311.03285)
++ [Punica: Multi-Tenant LoRA Serving](https://arxiv.org/abs/2310.18547)
+
+Now the vLLM has supported multi-lora, which integrated the `Punica` feature and related cuda kernels. See this [PR](https://github.com/vllm-project/vllm/pull/1804) for more. (2024-01-24 this PR has been merged into the main branch of vLLM)
+
+The following tutorial demonstrates how to deploy **a LLaMa model** with **multiple loras** on Triton Inference Server using the Triton's [Python-based](https://github.com/triton-inference-server/backend/blob/main/docs/python_based_backends.md#python-based-backends) [vLLM](https://github.com/triton-inference-server/vllm_backend/tree/main) backend.
+
+> Before you continue reading, it's important to note that all command-line instructions containing `<xx.yy>` in the document cannot be used directly by copying and pasting.
+>
+> `<xx.yy>` represents the Triton version, and you must specify the Triton version you want to use for the bash command to work.
+
+## Step 1: Start a docker container for triton-vllm serving
+
+**A docker container is strongly recommended for serving**, and this tutorial will only demonstrate how to launch triton in docker env.
+
+First, create a docker container using the NCG built for vllm serving:
+
+```bash
+# NOTICE: you must first cd to your vllm_workspace path outside the container.
+mkdir vllm_workspace && cd vllm_workspace
+
+sudo docker run --gpus all -it --net=host -p 8001:8001 --shm-size=12G \
+--ulimit memlock=-1 --ulimit stack=67108864 -v ${PWD}:/vllm_workspace \
+-w /vllm_workspace nvcr.io/nvidia/tritonserver:<xx.yy>-vllm-python-py3 \
+/bin/bash
+```
+
+**NOTICE:** the version of triton docker image should be configurated, here we use `<xx.yy>` to symbolize.
+
+Triton's vLLM container has been introduced starting from 23.10 release, and `multi-lora` experimental support was added in vLLM v0.3.0 release.
+
+> Docker image version `nvcr.io/nvidia/tritonserver:24.03-vllm-python-py3` or higher version is strongly recommended.
+
+---
+
+For **pre-24.03 containers**, the docker images didn't support multi-lora feature, so you need to replace that provided in the container `/opt/tritonserver/backends/vllm/model.py` with the most up to date version. Just follow this command:
```

Comment:
I believe this version(24.03) needs to updated to a version when this PR gets merged.  We have already code-frozen for 24.04 release. So, this would be 24.05 assuming PR gets merged before 24.05 code freeze. 
Can you change this version to 24.05 and add a TODO comment to check for the correct version?

Reply:
changed to 24.05, and added a TODO comment.